govcd/cse_type.go

```diff
@@ -0,0 +1,196 @@
+package govcd
+
+import (
+	semver "github.com/hashicorp/go-version"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"time"
+)
+
+// CseKubernetesCluster is a type for managing an existing Kubernetes cluster created by the Container Service Extension (CSE)
```

Comment:
Why these types are not in `./types`?
I would expect most of them to be in `./types`, and only  the usual "Outer" type to be in './govcd`

Reply:
The types from this file are "opinionated" and public, for the SDK users to consume them, whereas the `Capvcd` type from `./types` is the one owned/specified/managed by CSE developers (or in other words, it is "in VCD"), so the idea that I had is to put the ones that _we "own"_ in `./govcd`, and the one that _"VCD/CSE owns"_ in `/types`.

Or in other words, most of the types defined here would be "outer" types, but the difference is that here we don't do the typical wrap of "inner type + client", but instead we provide a set of computed properties that resemble what users see in UI.

I also chose to make the inner type private this time (`capvcdType *types.Capvcd`) as there is a lot of information there, hard to get/understand and sometimes outdated/doesn't reflect the actual cluster state (it is reflected accurately in YAML instead). The public properties reflect what users see in UI, they're picked from the actual YAML state in most cases and they should be enough, but we could discuss as maybe making `capvcdType` public would make sense to someone.

cc: @Didainius @lvirbalas

Reply:
> The types from this file are "opinionated" and public, for the SDK users to consume them, whereas the `Capvcd` type from `./types` is the one owned/specified/managed by CSE developers (or in other words, it is "in VCD"), so the idea that I had is to put the ones that _we "own"_ in `./govcd`, and the one that _"VCD/CSE owns"_ in `/types`.

I am good with this approach